src/Http/Exception/MissingControllerException.php

```diff
@@ -0,0 +1,43 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @since         3.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http\Exception;
+
+use Cake\Core\Exception\Exception;
+use Throwable;
+
+/**
+ * Missing Controller exception - used when a controller
+ * cannot be found.
+ */
+class MissingControllerException extends Exception
+{
+    /**
+     * @inheritDoc
+     */
+    protected $_messageTemplate = 'Controller class %s could not be found.';
+
+    /**
+     * Constructor.
+     *
+     * @param string|array $message Either the string of the error message, or an array of attributes
+     *   that are made available in the view, and sprintf()'d into Exception::$_messageTemplate
+     * @param int|null $code The code of the error, is also the HTTP status code for the error. Defaults to 404.
+     * @param \Throwable|null $previous the previous exception.
+     */
+    public function __construct($message, ?int $code = 404, ?Throwable $previous = null)
```

Comment:
I've looked why there is `404` here, but `500` in `ExceptionsTest::exceptionProvider()`.

It should be no `__construct()` here, but it should be `protected $_defaultCode = 404;` instead. With second argument as `null`, it fallbacks to `500`.

Core exceptions has mixed between `__construct()` and `protected $_defaultCode`, they should be probably normalized in next PR.